### PR TITLE
Refactoring functionality of article list using Redux

### PR DESCRIPTION
**Pull request by Carltesio**
# [PT Story](https://www.pivotaltracker.com/story/show/171811213)
## Changes proposed in this pull request:
* Implements Redux to make use of application state
* Improves the feature test by searching for the articles title and snippet in a correct order

## What I have learned working on this feature:
* Making better PRs
* Understand the basic Redux logic

## Packages/Gems added
- "redux": "^4.0.5"
- "redux-thunk": "^2.3.0"
-"react-redux": "^7.2.0"

### CODE REVIEW

File: src/components/DisplayArticles.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React, { Component } from "react";
+import { connect } from 'react-redux';
+
+class DisplayArticles extends Component {
+
+  render() {
+    let articleDisplay = this.props.articles.map(article => {
+      return (
+        <div id={'article-' + article.id} >
```
Comment (hmvitous) [inline src/components/DisplayArticles.jsx:9]: is this supposed to say  ```article-```?

Comment (janko101) [reply src/components/DisplayArticles.jsx:9]: Yes, it takes that string and adds the article id, so if it's the first article the id would be article-1...

File: src/state/reducers/rootReducer.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  switch (action.type) {
+    case 'GET_ARTICLE_DATA':
+      return {
+        ...state,
+        ...action.payload
+      }
+    default:
+      return state
+  }
+
```
Comment (janko101) [inline src/state/reducers/rootReducer.js]: Gonna remove empty space and push up

**Comment by tochman** [comment]
What does "Redux view articles" mean?

File: src/components/DisplayArticles.jsx (hunk 1 above)
Comment (hmvitous) [reply src/components/DisplayArticles.jsx:9]: Oh okay .
Thanks for clearing that up for me

File: src/index.js [hunk 3]
```diff
@@ -3,8 +3,15 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
-import axios from "axios";
+import { Provider } from 'react-redux';
+import configureStore from './state/store/configureStore'
```
Comment (pgaunitz) [inline src/index.js:9]: axios is not imported

**Review by pgaunitz — APPROVED** [review]

**Pull request merged**